mattermost.go

```diff
@@ -6,39 +6,339 @@ package main
+			flag.Usage()
+			os.Exit(1)
+		}
+
+		if len(flagPassword) == 0 {
+			fmt.Fprintln(os.Stderr, "flag needs an argument: -password")
+			flag.Usage()
+			os.Exit(1)
+		}
+
+		if len(flagPassword) < 5 {
+			fmt.Fprintln(os.Stderr, "flag invalid argument needs to be more than 4 characters: -password")
+			flag.Usage()
+			os.Exit(1)
+		}
+
+		c := &api.Context{}
+		c.RequestId = model.NewId()
+		c.IpAddress = "cmd_line"
+
+		var team *model.Team
+		if result := <-api.Srv.Store.Team().GetByName(flagTeamName); result.Err != nil {
+			l4g.Error("%v", result.Err)
+			flushLogAndExit(1)
+		} else {
+			team = result.Data.(*model.Team)
+		}
+
+		var user *model.User
+		if result := <-api.Srv.Store.User().GetByEmail(team.Id, flagEmail); result.Err != nil {
+			l4g.Error("%v", result.Err)
+			flushLogAndExit(1)
+		} else {
+			user = result.Data.(*model.User)
+		}
+
+		if result := <-api.Srv.Store.User().UpdatePassword(user.Id, model.HashPassword(flagPassword)); result.Err != nil {
+			l4g.Error("%v", result.Err)
+			flushLogAndExit(1)
+		}
+
+		os.Exit(0)
+	}
+}
+
+func flushLogAndExit(code int) {
+	l4g.Close()
+	time.Sleep(time.Second)
+	os.Exit(code)
+}
+
+var usage = `Mattermost commands to help configure the system
+Usage:
+
+    platform [options]
+
+    -config="config.json"             Path to the config file
+
+    -email="user@example.com"         Email address used in other commands
+
+    -password="mypassword"            Password used in other commands
+
+    -team_name="name"                 The team name used in other commands
+
+    -role="admin"                     The role used in other commands
+                                      valid values are
+                                        "" - The empty role is basic user 
+                                           permissions
+                                        "admin" - Represents a team admin and
+                                           is used to help adminsiter one team.
+                                        "system_admin" - Represents a system
+                                           admin who has access to all teams
+                                           and configuration settings.  This
+                                           role can only be created on the
+                                           team named "admin"
+
+    -create_team                      Creates a team.  It requres the -team_name
+                                      and -email flag to create a team.
+        Example:
+            platform -create_team -team_name="name" -email="user@example.com"
+
+    -create_user                      Creates a user.  It requres the -team_name,
+                                      -email and -password flag to create a user.
+        Example:
+            platform -create_user -team_name="name" -email="user@example.com" -password="mypassword"
+
+    -assign_role                      Assigns role to a user.  It requres the -team_name,
```

Comment:
The help text for assign role should point to the above explanation of -role. I didn't see it when I first read just assign_role.